nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -306,93 +290,48 @@ int ddxProcessArgument(int argc, char *argv[], int i)
 
       nxagentBinderLoop();
 
-      /*
-       * This will make an exit.
-       */
-
-      nxagentBinderExit(0);
+      nxagentBinderExit(0);      /* This will make an exit.*/
     }
     else
     {
-      /*
-       * Exit with an error.
-       */
-
-      nxagentBinderExit(1);
+      nxagentBinderExit(1);      /* Exit with an error.*/
     }
   }
 
   if (!strcmp(argv[i], "-display"))
   {
     if (++i < argc)
     {
-      strncpy(nxagentDisplayName, argv[i], 1023);
-
-      nxagentDisplayName[1023] = '\0';
-
+      strncpy(nxagentDisplayName, argv[i], sizeof(nxagentDisplayName) - 1);
```

Comment:
(changing those array to pointers is on my list already)

Well, _if_ we change nxagentDisplayName to be a pointer using a hardcoded value will also be wrong. So the situation does not get worse with my change. It only makes it easier to change that hardcoded value because it is now at only one location.

Besides, all this nullbyte stuffing will be gone once we are using pointers. We can even get rid of it now by using s(n)printf() instead of str(n)cpy().

Reply:
>  Well, *if* we change nxagentDisplayName to be a pointer using a hardcoded value will also be wrong. So the situation does not get worse with my change. It only makes it easier to change that hardcoded value because it is now at only one location.

Yes, that's why I recommend using a macro defining each length, e.g., `NXAGENTDISPLAYNAME_LENGTH = ...` in the header file and using that throughout the code instead of a hardcoded value.

Even if we change the variables to pure pointers, the situation wouldn't really change. We probably should bound the data length with pure pointers as well. Otherwise, `str(n)cpy` will crash on invalid input data. With bounding, we will in the worst case get `LEN-1` bytes of garbage data, but no actual crashes.

`sizeof(ptr)` doesn't work for pure pointers, so we'd have to change these sections again anyway.

I agree that dropping the hardcoded numbers is a good idea, but IMHO the best way (also taking into account further changes to pure pointers) is to use macros instead.